### PR TITLE
feat: support `exclude current_row`

### LINKED ISSUE #1387: Exclude the current row in window aggregation
**Is your feature request related to a problem? Please describe.**
In time window aggregation, we would like to to have support for us to exclude the current row in aggregation. This is useful because in some cases, the current row only serves as a pivot to offer timestamps to identify a certain event. Any other  data of this row is not relevant for the aggregation.

**Describe the solution you'd like**
`SELECT count(*) OVER w1  as w1_col2_sum FROM t1
WINDOW w1 as (PARTITION BY user_id ORDER BY time ROWS_RANGE BETWEEN 1s PRECEDING AND CURRENT ROW EXCLUDE CURRENT ROW)`

**Describe alternatives you've considered**
We can achive our goal now in a very unintuitive way:
First we add 1 to time stamp column(in milliseconds)
Then we use "1 preceeding" to exclude the added 1 millisecond, as follows:

`SELECT count(*) OVER w1  as w1_col2_sum FROM t1
WINDOW w1 as (PARTITION BY user_id ORDER BY time ROWS_RANGE BETWEEN 1s PRECEDING AND 1 PRECEDING)`

**Additional context**

### PR DESCRIPTION
**Pull request by aceforeverd**
close #1387

Here's the overviews

- Overview
	- `window ( ROWS_RANGE .. [ 0s preceding | current row ] exclude current_row)` -> special handling
		- actually a mixed type window: frame start with range offset, and frame end as rows offset
	- `window (ROWS .. [ 0 preceding | current row ] exclude current_row)` -> `window (ROWS .. 0 open preceding)`
	- `lag over ( .. exclude current_row)` -> no effect
	- other case -> no effect
- plan nodes
   new field `exclude_current_row_` added to those nodes
	- logical nodes
		- `FrameNode` -> set only by physical transformer
		- `WindowDefNode` -> set by planner
		- `WindowPlanNode`
	- physical nodes
		- `PhysicalWindowAggrerationNode`
		- `PhysicalRequestUnionNode`
- runner
	- `WindowAggRunner` batch mode
	- `RequestUnionNode` (batch-)request mode
- codegen
	- `InnerRowsRangeList` -> for (batch-)request mode, construct the `InnerRowsRangeList(1, rows_end)` for `rows_range ( .. os preceding exclude current_row)`
- iterator
	- the `HistoryWindow` table handler for batch mode.

### CODE REVIEW

**Comment by aceforeverd** [comment]
[exclude current row](https://trello.com/c/2KPIBLFI/105-exclude-current-row)

File: hybridse/src/vm/runner.cc [hunk 1]
```diff
@@ -3135,11 +3143,22 @@ std::shared_ptr<TableHandler> RequestUnionRunner::RequestUnionWindow(
         }
         rows_start_preceding = window_range.start_row_;
         max_size = window_range.max_size_;
+
+        // TODO(ace): should optimized in codegen window builder
```
Comment (aceforeverd) [inline hybridse/src/vm/runner.cc]: TODO done

**Comment by aceforeverd** [comment]
TODO:
- pre-agg
- ~~openmldb-batch~~

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: hybridse/include/codec/list_iterator_codec.h [hunk 2]
```diff
@@ -459,10 +495,33 @@ class InnerRowsList : public ListV<V> {
...
+// |<--------  iterator goes
+template <class V>
+class InnerRowsRangeList : public ListV<V> {
+ public:
+    InnerRowsRangeList(ListV<V> *root, uint64_t start_rows, uint64_t end_range)
+        : ListV<V>(), root_(root), start_rows_(start_rows), end_range_(end_range) {}
+    ~InnerRowsRangeList() override {}
+
+    std::unique_ptr<ConstIterator<uint64_t, V>> GetIterator() override {
+        return std::make_unique<InnerRowsRangeIterator<V>>(std::move(root_->GetIterator()), start_rows_, end_range_);
```
Comment (zhanghaohit) [inline hybridse/include/codec/list_iterator_codec.h]: `std::move` is not necessary?

Comment (aceforeverd) [reply hybridse/include/codec/list_iterator_codec.h]: reasonable !

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**